### PR TITLE
[stable/gocd] Add a post start lifecycle hook for GoCD server to preconfigure the server

### PR DESCRIPTION
**Pull request by varshavaradarajan**
**What this PR does / why we need it**:

* This makes it easier for getting started with GoCD for new users. 
* Added a changelog to keep track of releases and changes for each.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/gocd/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "gocd.fullname" . }}
+  labels:
+    app: {{ template "gocd.fullname" . }}
```
Comment (unguiculus) [inline stable/gocd/templates/configmap.yaml]: `app: {{ template "gocd.name" . }}`

File: stable/gocd/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### 1.0.4
```
Comment (unguiculus) [inline stable/gocd/CHANGELOG.md:3]: How about 1.0.5 in the changelog?

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by varshavaradarajan** [comment]
@unguiculus - done. Thank you. 

Apart from being in the OWNERS file, how do we ensure that we can maintain and merge PRs for the GoCD helm charts ourselves?

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
@varshavaradarajan You must become a member of the Kubernetes Github org.

See https://github.com/kubernetes/charts#owning-and-maintaining-a-chart

I can sponsor that.

**Comment by varshavaradarajan** [comment]
@unguiculus - thank you. Every release, we want to bump up the app version to be the latest GoCD version. We are thinking of setting up CI for that. So, I was thinking if it was possible for the ci bot user to have commit access to the chart as well. Is that possible?

**Pull request merged**